src/Reservation.php

```diff
@@ -181,6 +182,94 @@ public function prepareInputForAdd($input)
         return parent::prepareInputForAdd($input);
     }
 
+    public static function handleAddForm(array $input)
+    {
+        $rr = new self();
```

Comment:
You should exit from method here if there is nothing to add, it would prevent code to be nested to many if blocks.

```suggestion
        if (
            !isset($input['resa']["begin"])
            || !isset($input['resa']["end"])
            || !isset($input['items'])
            || !is_array($input['items'])
            || count($input['items']) === 0
            || !isset($input['users_id'])
        ) {
            return;
        }

        $rr = new self();
```